psqlgraph/node.py

```diff
@@ -151,11 +151,22 @@ def traverse(self, mode="bfs", max_depth=None, edge_pointer="in", edge_predicate
                 edge_pointer=edge_pointer,
                 max_depth=max_depth
             )
+
+        # max depth is ignored for dfs
```

Comment:
dfs can still be depth limited.  since it's an argument of `traverse`, the algorithms we implement here should also support it.

Reply:
Not sure how to solve depth of this situation:

A ---> B ---> D ---> E
  \       /
   \-> C /